include/boost/geometry/algorithms/detail/overlay/follow_linear_linear.hpp

```diff
@@ -304,7 +326,14 @@ class follow_linestring_linear_linestring
                                oit);
         }
 
-        BOOST_ASSERT( enter_count == 0 );
+#if ! defined(BOOST_GEOMETRY_OVERLAY_NO_THROW)
+        if (enter_count != 0)
+        {
+            throw inconsistent_turns_exception();
+        }
+#else
+        BOOST_ASSERT(enter_count == 0);
+#endif
 
```

Comment:
OK for me.
We might think of a macro which implements this behaviour in a sort of generic way, for example
```
BOOST_GEOMETRY_THROW_OR_ASSERT(enter_count == 0, 
  inconsistent_turns_exception, 
  "Enter count: " << enter_count);
```
so the third argument is a streamable thing part of the exception